app/models/builders/pagination.js

```diff
@@ -0,0 +1,99 @@
+const { pagination } = require('../../constants/search')
+
+const calcMidSectionStart = (numOfPages, currentPage) => {
+  const startPos = currentPage - 1
+  if (startPos <= 1) {
+    return 2
+  }
+  if (startPos + 3 >= numOfPages) {
+    return Math.max(numOfPages - 3, 2)
+  }
+  return startPos
+}
+
+const stripPageParam = (url) => {
+  const foundPos = url.indexOf('&page=')
+  if (foundPos === -1) {
+    return url
+  }
+  return url.substr(0, foundPos)
+}
+
+const getPage = (baseUrl, pageNum, currentPage) => (
+  {
+    number: pageNum,
+    href: `${baseUrl}&page=${pageNum}`,
+    current: currentPage === pageNum
+  }
+)
+
+const addPageNumbers = (numOfPages, currentPage, baseUrl) => {
+  const pages = [getPage(baseUrl, 1, currentPage)]
+
+  const midSectionStart = calcMidSectionStart(numOfPages, currentPage)
+  const midSectionEnd = midSectionStart + 2
+
+  if (midSectionStart > 2) {
+    pages.push({ ellipsis: true })
+  }
+
+  for (let i = midSectionStart; i < midSectionEnd + 1; i++) {
+    if (i < numOfPages) {
+      pages.push(getPage(baseUrl, i, currentPage))
+    }
+  }
+
+  if (midSectionEnd < numOfPages - 1) {
```

Comment:
won't make any difference to but brackets around the right hand side would make this clearer.